aiida/orm/implementation/entities.py

```diff
@@ -0,0 +1,451 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Classes and methods for backend non-specific entities"""
+import abc
+import typing
+
+from aiida.orm.implementation.utils import clean_value, validate_attribute_extra_key
+
+__all__ = ('BackendEntity', 'BackendCollection', 'EntityType', 'BackendEntityAttributesMixin', \
```

Comment:
Just small reference for the future (don't need to adapt it here): the slash is not necessary when you are in between parentheses.